### ISSUE TITLE
Azure Storage provider buffered classes fail to load blob-specific client

### ISSUE THREAD

**Issue by stewartadam** [Accelize/pycosio#3]
Operations calling [AzureBlobBufferedIO](https://github.com/Accelize/pycosio/blob/master/pycosio/storage/azure_blob.py#L306) fail because it's `self._client` property appears to be derived from [_AzureBlobSystem._get_client](https://github.com/Accelize/pycosio/blob/70fa5e2444395a980aaa1b463ad5479043ba7930/pycosio/storage/azure_blob.py#L54) (i.e., a dict of the blob service classes) and not [AzureBlobRawIO._client](https://github.com/Accelize/pycosio/blob/70fa5e2444395a980aaa1b463ad5479043ba7930/pycosio/storage/azure_blob.py#L256) as the code seems to expect (ie., the specific blob service implementation for the given blob type), so a `copyfile()` attempt results in:
```
Traceback (most recent call last):
  File "cloud-transfer.py", line 221, in <module>
    upload(path, url)
  File "cloud-transfer.py", line 196, in upload
    handlers[scheme](path, url)
  File "cloud-transfer.py", line 137, in upload_azure_blob
    pycosio.copyfile(path, url)
  File "/Users/stewartadam/devel/virtualenvs/filetransfer-2tnmA_nJ/lib/python3.7/site-packages/pycosio/_core/functions_shutil.py", line 142, in copyfile
    _copy(src, dst, src_is_storage, dst_is_storage)
  File "/Users/stewartadam/devel/virtualenvs/filetransfer-2tnmA_nJ/lib/python3.7/site-packages/pycosio/_core/functions_shutil.py", line 71, in _copy
    copyfileobj(fsrc, fdst, buffer_size)
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "/Users/stewartadam/devel/virtualenvs/filetransfer-2tnmA_nJ/lib/python3.7/site-packages/pycosio/_core/functions_io.py", line 79, in cos_open
    yield wrapped
  File "/Users/stewartadam/devel/virtualenvs/filetransfer-2tnmA_nJ/lib/python3.7/site-packages/pycosio/_core/io_buffered.py", line 118, in close
    self._flush()
  File "/Users/stewartadam/devel/virtualenvs/filetransfer-2tnmA_nJ/lib/python3.7/site-packages/pycosio/storage/azure_blob.py", line 360, in _flush
    self._client.put_block, block=self._get_buffer(),
AttributeError: 'dict' object has no attribute 'put_block'
```

**Comment by stewartadam** [comment]
For repro -- I am calling:

```
parameters = {
    'account_name': 'my_account',
    'account_key': 'my_key
}
pycosio.mount(storage='azure_blob', storage_parameters=parameters)
pycosio.copyfile(local_path, blob_uri)
```

After having manually applied the 'fixes' in #2 (if I can get this all working I'll submit an PR)

**Comment by xlz-jgoutin** [comment]
Fixed in PR #5.
